TeslaLib/TeslaClient.cs

```diff
@@ -113,7 +113,81 @@ public TeslaClient(string email, string teslaClientId, string teslaClientSecret,
 
             TeslaAuthHelper = authHelper ?? new TeslaAuthHelper(FlexChargingUserAgent, region);
         }
-        public IOAuthTokenStore OAuthTokenStore { get; set; }
+        public IOAuthTokenStore OAuthTokenStore { get; set; } 
+        public async Task LoginUsingTokenStoreAsync(LoginToken loginToken, bool forceRefreshOlderThanToday = false)
+        {
+            if (string.IsNullOrWhiteSpace(Email))
+            {
+                throw new ArgumentNullException(nameof(Email));
+            }
+            var token = await TokenStoreForThisInstance.GetTokenAsync(Email);
+
+            if (token == null)
+            {
+                await TokenStoreForThisInstance.AddTokenAsync(Email, loginToken);
+
+                token = await TokenStoreForThisInstance.GetTokenAsync(Email);
+            }
+            else
+            {
+                await TokenStoreForThisInstance.UpdateTokenAsync(Email, loginToken);
+
+                token = await TokenStoreForThisInstance.GetTokenAsync(Email);
+            }
+
+            TimeSpan expirationTimeFromNow = token.ExpiresUtc - DateTime.UtcNow;
+
+            if (expirationTimeFromNow < TokenExpirationRenewalWindow || forceRefreshOlderThanToday)

```

Comment:
The forceRefrehsOlderThanToday parameter is no longer named properly when token livetime is 8 hours.  We should probably call this forceRefresh.

Reply:
Thanks, Done